### ISSUE TITLE
insert_and_get returns an incomplete model (e.g model._state.db is not set correctly)

### ISSUE THREAD

**Issue by elnygren** [SectorLabs/django-postgres-extra#25]
**Reproduce:**

```
# assumptions:
# * MyModel.users is an m2m field to User
# * user = User instance

my_obj = MyModel.objects.upsert_and_get(conflict_target=['id'],fields=data)
my_obj.users.add(user)  # will throw
```

**Exception:**

```
ValueError: Cannot add "<MyModel: ...>": instance is on database "default", value is on database "None"
```

**Problem in sources:**

In, `PostgresQuerySet` the `insert_or_get` used in `upsert_and_get` returns the following:

```
return self.model(**model_init_fields)
```

(for example, see https://github.com/SectorLabs/django-postgres-extra/blob/859b37a94fce1303bfd2943a49d0bfaba53754b5/psqlextra/manager/manager.py#L217)

In reality, initialising a Model instance requires a bit more magic:

https://github.com/django/django/blob/a51c4de1945be2225f20fad794cfb52d8f1f9236/django/db/models/base.py#L633
https://github.com/django/django/blob/a51c4de1945be2225f20fad794cfb52d8f1f9236/django/db/models/base.py#L714

Note that in Django, `MyModel.objects.create` uses MyModel.save (which uses MyModel.save_base). Things like `MyModel._state.db` are set correctly in this situation.

**Quick and dirty workaround**

`my_obj._state.db = 'default'` before doing `my_obj.users.add(user)`

**Proposed fix**

Make sure that every method that returns the Model instance properly initialises the instance. This will probably require some digging around in the Django sources (especially the two methods linked above).

**Comment by Photonios** [comment]
Wow! Thanks for the detailed bug report! I'll have a more detailed look tonight :)

**Comment by Photonios** [comment]
Sorry it took so long to get to this, been busy. Managed to free some time during my holiday in the Carribbean :-)

**Comment by Amoki** [comment]
I have the same problem with version 1.20
I need to manually `save()` the object after the `insert_or_get()` call which seems to be a bad idea in the context of an upsert.

**Comment by Photonios** [comment]
@Amoki could you post an isolated example of how that happens? So I can easily write a test for it and fix it.

**Comment by LegerG** [comment]
Hi, I have an isolated example : 

```python
for label in labels:
            l = (
                Label.objects
                .on_conflict(['label'], ConflictAction.UPDATE)
                .insert_and_get(label=label)
            )
            l.save()  # we must call .save()
            topic.labels.add(l)
```

**Comment by Photonios** [comment]
Can you also show how the model ‘Label’ looks?

**Comment by LegerG** [comment]
```python
class Topic(PostgresModel):
    labels = models.ManyToManyField('Label', blank=True)

class Label(PostgresModel):
    label = models.CharField(max_length=255, null=True, blank=True, unique=True)

    def __str__(self):
        return self.label
```

**Comment by Photonios** [comment]
Fixed in: https://github.com/SectorLabs/django-postgres-extra/commit/82526a5feaa7d450f7108d9d35209b0745e28162

Will be released in v2.0 soon.
